### PR TITLE
:seedling: Added Upload Artifact Condition to Main

### PR DESCRIPTION
**Pull request by pallav-trilio**

### CODE REVIEW

File: .github/workflows/plugin-packages.yml [hunk 1]
```diff
@@ -143,6 +145,16 @@ jobs:
       - name: Test Log-Collector Plugin Locally
         run: make test-log-collector-plugin-locally
 
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@master
```
Comment (msurudwad) [inline .github/workflows/plugin-packages.yml:149]: how are you ensuring this step runs for `main` branch only, not for PRs & tags?

**Review by msurudwad — CHANGES_REQUESTED** [review]

File: .github/workflows/plugin-packages.yml [hunk 2]
```diff
@@ -143,6 +145,18 @@ jobs:
...
 
+      - name: Set up Cloud SDK
+        if: ${{ github.ref == 'refs/heads/main' }}
+        uses: google-github-actions/setup-gcloud@master
+        with:
+          project_id: ${{ secrets.GCS_PROJECT_ID }}
+          service_account_key: ${{ secrets.GCS_ARTIFACTS_CREDENTIALS }}
+          export_default_credentials: true
+
+      - name: Upload Preflight Plugin Packges to github artifact
```
Comment (msurudwad) [inline .github/workflows/plugin-packages.yml:156]: we are not uploading to github artifact right? change the `name`

**Review by msurudwad — COMMENTED** [review]
LGTM

File: .github/workflows/plugin-packages.yml (hunk 2 above)
Comment (pallav-trilio) [reply .github/workflows/plugin-packages.yml]: done

**Review by madhurtrilio — APPROVED** [review]

**Pull request merged**